libpysal/graph/_set_ops.py

```diff
@@ -34,46 +34,42 @@ def __xor__(self, other):
         return symmetric_difference(self, other)
 
     def __iand__(self, other):
-        raise TypeError("weights are immutable")
+        raise TypeError("Graphs are immutable")
 
     def __ior__(self, other):
-        raise TypeError("weights are immutable")
+        raise TypeError("Graphs are immutable")
 
     def __len__(self):
         return self.n_edges
 
 
-# TODO: performance test the pandas implementation
 def intersects(left, right):
     """
-    A full table join is unnecessary here, but I'm not sure if
-    it would be faster? pandas joins are very fast, even for big tables,
-    but this can do early termination at the first intersection.
-
-    Maybe we use a heuristic to pick the fastest code path? It's also
-    very easy to do early termination in cython/numba using the row,col array.
+    Returns True if left and right share at least one link, irrespective of weights
+    value.
     """
-    for left_focal, left_neighbors in left.neighbors.keys():
-        right_neighbors = right.neighbors.get(left_focal)
-        if right_neighbors is not None:
-            if set(right_neighbors).intersects(set(left_neighbors)):
-                return True
+    if left._adjacency.index.isin(right._adjacency.index).any():
+        return True
     return False
-    return not left.adjacency.join(
-        right.adjacency, on=("focal", "neighbor"), how="inner"
-    ).empty
 
 
 def intersection(left, right):
     """
-    Keep only links that are in both left and right Graph objects.
+    Returns a binary weights object, that includes only those neighbor pairs that exist
```

Comment:
This should be `Graph`, correct?

```suggestion
    Returns a binary Graph object, that includes only those neighbor pairs that exist
```